### PR TITLE
Search for products

### PR DESCRIPTION
**Pull request by antoalconti**
#### Board:
* [As a user, I should be able to search for products](https://www.notion.so/rootstrap/15530953f92c4702a8008136b3b3398d?v=151b8fc31613405eb6358cffdaff6628&p=762d10779e7041e0b589a001e0c88629&pm=s)
---
#### Description:

---
#### Notes:
*
---
#### Tasks:

https://github.com/user-attachments/assets/ed10fba9-6752-4849-80b7-7fbedaa63c7c

- [x] Add each element in this format
---
#### Risk:
*
---

### CODE REVIEW

File: app/views/layouts/_navbar.html.erb [hunk 1]
```diff
@@ -1,10 +1,17 @@
-<div class="flex bg-black h-24 md:h-14 px-6 gap-10 justify-around m-auto" >
-    <%= link_to "index", class: 'h-10 w-48 my-auto' do %>
+<div class="flex bg-black px-6 gap-10 justify-around m-auto lg:h-24 h-14 sticky items-center" >
+    <%= link_to root_path, class: 'my-auto' do %>
       <%= image_tag("logo") %>
     <% end %>
-    <%= link_to "shopping_cart_path", class: 'h-10 w-18 p-2 border inline-flex text-white rounded my-auto' do %>
-      <span class="font-sans text-sm text-white hidden md:inline">Shopping Cart</span>
-      <%= image_tag("cart") %>
+    <div class="flex flex-col md:flex-row items-start md:items-center justify-betwee w-full">
```
Comment (santiagodiaz) [inline app/views/layouts/_navbar.html.erb:5]: not sure if this justify-between was applied, maybe you can try this?
```suggestion
    <div class="flex flex-col md:flex-row items-start md:items-center justify-between w-full">
```

**Review by santiagodiaz — APPROVED** [review]

File: app/javascript/controllers/search_product_controller.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  submit() {
+    clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.formTarget.requestSubmit();
+    }, 300);
+  }
+
+  get formTarget() {
+    return this.targets.find("form");
+  }
```
Comment (santib) [inline app/javascript/controllers/search_product_controller.js:19]: I don't think this is needed 🤔 

https://stimulus.hotwired.dev/reference/targets

File: app/views/layouts/_navbar.html.erb [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+    <%= form_with url: products_path, method: :get, data: { controller: 'search-product', search_product_target: 'form' } do |f| %>
```
Comment (santib) [inline app/views/layouts/_navbar.html.erb:6]: not for now, but I think it's a bit strange that generic code that lives in the layout, always hit a very specific route like `products_path`

File: app/views/layouts/_navbar.html.erb [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+      <%= search_field_tag :q, params[:q], class: "text-gray-800", placeholder: "Search...", data: { action: "input->search-product#submit" } %>
     <% end %>
-</div>
 
+    </div>
+
+
+   <%= link_to "shopping_cart_path", class: 'p-3 px-5 border rounded mx-auto inline-flex items-center' do %>
+      <span class="font-sans text-sm text-white hidden lg:inline whitespace-nowrap">Shopping Cart</span>
```
Comment (santib) [inline app/views/layouts/_navbar.html.erb:14]: we should always `i18n` any user facing text

**Comment by santib** [comment]
QA 


https://github.com/user-attachments/assets/1cf72120-e780-4564-bc8e-f5acaab880d7

**Review by santib — APPROVED** [review]

**Pull request merged**